perfkitbenchmarker/analyze.py

```diff
@@ -0,0 +1,202 @@
+
+  return interval_start - first_start, last_end - interval_end
+
+
+def FullyInInterval(start_times, durations, interval_start, interval_duration):
+  """Compute which records are completely inside an interval.
+
+  Args:
+    start_times: a pd.Series of POSIX timestamps, as floats
+    durations: a pd.Series of durations in seconds, as floats
+    interval_start: the POSIX timestamp of the interval start, as a float
+    interval_duration: the duration of the interval in seconds, as a float
+
+  start_times and durations must have matching indices. Each
+  (start_time, duration) pair is considered a record of an operation
+  that began at the given start time and continued for the given
+  duration.
+
+  Returns: a pd.Series of booleans. An element is True if the
+    corresponding record lies completely within the given interval,
+    and false otherwise.
+  """
+
+  interval_end = interval_start + interval_duration
+  record_ends = start_times + durations
+
+  return (start_times >= interval_start) & (record_ends <= interval_end)
+
+
+def AllStreamsThroughputStats(durations, sizes, stream_ids,
+                              num_streams, interval_duration):
+  """Compute the net throughput of multiple streams doing operations.
+
+  Args:
+    durations: a pd.Series of durations in seconds, as floats.
+    sizes: a pd.Series of bytes.
+    stream_ids: a pd.Series of any type.
+    num_streams: int. The number of streams.
+    interval_duration: a float. The time all streams were active, in seconds.
+
+  durations, sizes, and stream_ids must have matching indices. This
+  function computes the per-stream net throughput (sum of bytes
+  transferred / total transfer time) and then adds the results to find
+  the overall net throughput. Operations from the same stream cannot
+  overlap, but operations from different streams may overlap.
+
+  Returns: a tuple of
+    - The net throughput of all streams, excluding times they were inactive.
+    - The throughput of all streams, including times they were inactive.
+    - The total time that all streams were inactive.
+    - The total time all streams were inactive, as a proportion of the
+      total benchmark time.
+  """
+
+  total_bytes_by_stream = sizes.groupby(stream_ids).sum()
+  active_time_by_stream = durations.groupby(stream_ids).sum()
+  total_overhead = interval_duration * num_streams - active_time_by_stream.sum()
+
+  return ((total_bytes_by_stream / active_time_by_stream).sum(),
+          total_bytes_by_stream.sum() / interval_duration,
+          total_overhead,
+          total_overhead / (interval_duration * num_streams))
+
+
+def SummaryStats(series, name_prefix=''):
+
+  """Compute some summary statistics for a series.
+
+  Args:
+    series: a pd.Series of floats.
+    name_prefix: if given, a prefix for the statistic names.
+
+  Returns: a pd.Series with summary statistics.
+  """
+
+  # Percentiles 0, 1, 2, ..., 100
+  percentiles = range(0, 101, 1)
```

Comment:
it's an overkill to compute percentiles to the ones digit, e.g.: p33, p34, p35 and p36, etc, no one is going to use it.

Also, this function functionally duplicates with the percentile calculator that we already have, can you please use that calculator instead of introduce a new one here?